src/utilities/resolveRoutes.spec.ts

```diff
@@ -90,6 +90,53 @@ describe('resolveRoutes', () => {
     expect(resolvedChild?.path).toBe(`${parentRoute.path}/:accountId`)
   })
 
+  test.each([
+    ['id=:accountId'],
+    [query('id=:accountId', { accountId: Number })],
+  ])('always combines query into return query', (query) => {
+    const childRoute: Route = {
+      name: 'new-account',
+      path: '/new-account',
+      query: 'handle=:?handle',
+      component,
+    }
+
+    const parentRoute: Route = {
+      name: 'accounts',
+      path: '/accounts',
+      query,
+      children: [childRoute],
+    }
+
+    const routes = resolveRoutes([parentRoute])
+    const resolvedChild = routes.find(route => route.name === childRoute.name)
+
+    expect(resolvedChild?.query).toBe('id=:accountId&handle=:?handle')
+  })
+
+  test('always combines params into return params', () => {
+    const childRoute: Route = {
+      name: 'new-account',
+      path: '/new-account',
+      query: 'handle=:handle',
+      component,
+    }
+
+    const parentRoute: Route = {
+      name: 'accounts',
+      path: ':workspaceId/accounts',
+      query: 'start=?:startDate',
+      children: [childRoute],
+    }
+
+    const routes = resolveRoutes([parentRoute])
+    const resolvedChild = routes.find(route => route.name === childRoute.name)
+
+    expect(resolvedChild?.params).toMatchObject({ workspaceId: [String], startDate: [String], handle: [String] })
+    // check
+    expect(Object.keys(resolvedChild?.params ?? {})).toMatchObject(['workspaceId', 'startDate', 'handle'])
```

Comment:
Won't the first expect verify the keys are correct?

Reply:
yes, but not in the correct order. that comment was supposed to say "check order"

Reply:
Why do we care about the order of the keys in the object?

Reply:
Because the order of params here is the order they will be evaluated when trying to match a url. Parent path params, then child path params, then parent query params, then child query params

Reply:
That only matters within same named params though right?

Reply:
The object keys are not going to be in that order. They're alphabetical I believe.

Reply:
Ah you're right. I was confusing the two. I'll remove the line that checks order.